Source/ACE.DatLoader/Entity/AnimationHooks/CreateParticleHook.cs

```diff
@@ -0,0 +1,33 @@
+﻿using ACE.Entity;
+
+namespace ACE.DatLoader.Entity.AnimationHooks
+{
+    public class CreateParticleHook : IHook
+    {
+        public uint EmitterInfoId { get; set; }
+        public uint PartIndex { get; set; }
+        public Position Offset { get; set; }
+        public uint EmitterId { get; set; }
+
+        public static CreateParticleHook ReadHookType(DatReader datReader)
+        {
+            CreateParticleHook hook = new CreateParticleHook();
+            hook.EmitterInfoId = datReader.ReadUInt32();
+
+            hook.PartIndex = datReader.ReadUInt32();
+
+            Position p = new Position();
+            p.PositionX = datReader.ReadSingle();
```

Comment:
Consider making an extension method for Position.Read(DatReader datReader) to reduce redundancy in reading these 7 singles.  You can put this class in the DatLoader project somewhere, and then call it like you would any other method on Position.

```cs
public static class PositionExtensions
{
    public static Position ReadPosition(DatReader datReader)
    {
        Position p = new Position();
        p.Read(datReader);
        return p;
    }

    public static void Read(this Position p, DatReader datReader)
    {
        p.PositionX = datReader.ReadSingle();
        p.PositionY = datReader.ReadSingle();
        p.PositionZ = datReader.ReadSingle();
        p.RotationW = datReader.ReadSingle();
        p.RotationX = datReader.ReadSingle();
        p.RotationY = datReader.ReadSingle();
        p.RotationZ = datReader.ReadSingle();
    }
}
```

You could then call:
```cs
hook.Offset = PositionExtensions.ReadPosition(datReader);
```